packages/runtime/src/runtimes/react/components/DocumentRoot.tsx

```diff
@@ -0,0 +1,25 @@
+import { Ref, forwardRef, memo } from 'react'
+import { type Document } from '../../../state/react-page'
+import { ElementImperativeHandle } from '../element-imperative-handle'
+import { useDocument } from '../hooks/use-document'
+import { FallbackComponent } from '../../../components/shared/FallbackComponent'
+import { Document as DocumentComponent } from './Document'
+
+type Props = {
+  rootDocument: Document
+}
+
+export const DocumentRoot = memo(
+  forwardRef(function DocumentRoot(
+    { rootDocument }: Props,
+    ref: Ref<ElementImperativeHandle>,
+  ): JSX.Element {
+    const document = useDocument(rootDocument.key) ?? rootDocument
```

Comment:
@migueloller We don't register anything here, but rather try to retrieve the (latest version of) document from the state by its key and fall back to the initial version if there is none (yet). 

If that's what you meant, can you clarify your comment about "keeping concerns separate"?

Reply:
I see. I wonder if we should have `useDocument` take the "fallback" document as an argument similar to how `useSyncExternalStore`, `useState`, `useActionState`, and other React hooks take a default value as a parameter.

As for "keep concerns separate" I thought that `useDocument` was registering the document (I see now that's impossible since it just takes a key) so I was alluding to separating registration from reading.

Where does registration happen now?